### PR TITLE
Mod linked list interface

### PR DESCRIPTION
**Pull request by neithernut**
This PR adapts the linked list interface to fit in the layer model described in issue #103.

### CODE REVIEW

**Comment by neithernut** [comment]
I forgot to adapt the tests. I will do so now.

File: src/libreset/ll.c [hunk 1]
```diff
@@ -40,74 +42,52 @@ ll_destroy(
...
 ll_insert(
     struct ll* ll,
-    struct ll_element* e
+    void* data,
+    struct r_set_cfg* cfg
 ) {
-    if (!ll || !e) {
+    // check whether the lement is present or not
+    struct ll_element** it=&ll->head;
+    while (*it) {
```
Comment (matthiasbeyer) [inline src/libreset/ll.c:52]: while(true) here.

File: src/libreset/ll.c [hunk 2]
```diff
@@ -40,74 +42,52 @@ ll_destroy(
...
-    }
-
-    for (iter = ll->head; iter; iter = iter->next) {
-        if (iter == del) {
-            last->next = iter->next;
-            free(iter);
-            break;
+    // iterate over all the elements
+    while (*iter) {
+        // check whther we have found the element to remove
```
Comment (matthiasbeyer) [inline src/libreset/ll.c:82]: typo

File: src/libreset/ll.h [hunk 3]
```diff
@@ -61,23 +63,13 @@ struct ll {
  *
  * Removes the linked list from the memory but does not remove the referenced
  * elements.
+ *
+ * @warning the function may crashed if either of the arguments is NULL
```
Comment (matthiasbeyer) [inline src/libreset/ll.h]: typo here "may crashed" -> "may crash"

File: src/libreset/ll.h [hunk 4]
```diff
@@ -86,24 +78,28 @@ ll_insert(
...
 
 /**
- * Delete a struct ll_element from the linked list
+ * Delete an item from the list
  *
  * @memberof ll
  *
- * @return the struct ll object which was passed or NULL on failure
+ * @return the struct ll object which was passed
+ * @warning the function may crash if either of the arguments is NULL
```
Comment (matthiasbeyer) [inline src/libreset/ll.h:96]: Please move the `@warning` above the `@return` - the `@return` should always be the last documentation.

**Comment by neithernut** [comment]
Please review again and let me know if there are any issues left to fix. I will then squash.

File: src/libreset/ll.c [hunk 5]
```diff
@@ -25,89 +25,77 @@
...
-    if (!ll || !e) {
-        return NULL;
+    // check whether the lement is present or not
+    struct ll_element** it=&ll->head;
+    while (*it) {
+        if (cfg->cmpf((*it)->data, data)) {
+            return 0;
+        }
+
+        it=&(*it)->next;
```
Comment (matthiasbeyer) [inline src/libreset/ll.c]: How could this pass the pre-commit hook, there are missing spaces around `=` here!?

**Comment by matthiasbeyer** [comment]
Fix the missing space comment and then ACK for squashing!

File: src/libreset/ll.c (hunk 5 above)
Comment (neithernut) [inline src/libreset/ll.c]: Somehow the hook doesn't work.

**Comment by neithernut** [comment]
I did not clean the history as exessive as intended but it should suffice.

**Pull request merged**